fix: TestWorkflowArchiveServiceList and TestWorkflowServiceListArchived panic on re-run

## Summary

`TestWorkflowArchiveServiceList` and `TestWorkflowServiceListArchived` in `test/e2e/argo_server_test.go` panic with a nil pointer dereference on test re-run, causing CI flakiness.

## Root Cause

Both tests create workflows inside `s.Run()` subtests and store the result in variables (`bobWf`, `aliceWf`). However, they extract fields like `uid` and `name` from these variables **outside** the `s.Run()` closures:

```go
var bobWf *httpexpect.Value
s.Run("CreateArchivedBobWf", func() {
    bobWf = s.e().POST(...).Expect().Status(200).JSON()
})
var uidBobWf = bobWf.Path("$.metadata.uid")  // panics on re-run
```

When Go's test framework re-runs a failed test, it re-executes the parent test function but skips subtests that already passed. The `s.Run` closure does not execute on re-run, so `bobWf` stays `nil`, and the `.Path()` call panics.

## CI Evidence

This test fails consistently on re-run in CI across multiple runs on `main`:
- https://github.com/argoproj/argo-workflows/actions/runs/22465028354 (`test-api, mysql, true`)
- https://github.com/argoproj/argo-workflows/actions/runs/22454215475 (`test-api, mysql, true`)

The panic trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
github.com/gavv/httpexpect/v2.(*Value).Path(0x0, ...)
    value.go:169
github.com/argoproj/argo-workflows/v4/test/e2e.(*ArgoServerSuite).TestWorkflowArchiveServiceList(...)
    argo_server_test.go:1245
```

## Fix

Move the field extraction (`uid`, `name`) inside the `s.Run()` closures so the variables are always set in the same scope as the workflow creation. This ensures they are populated on every run, including re-runs.

**1 file changed, +20/-28 lines.**

--- a/test/e2e/argo_server_test.go
+++ b/test/e2e/argo_server_test.go
@@ -1029,9 +1029,9 @@ func (s *ArgoServerSuite) TestWorkflowService() {
 }
 
 func (s *ArgoServerSuite) TestWorkflowServiceListArchived() {
-	var bobWf *httpexpect.Value
+	var uidBobWf, nameBobWf string
 	s.Run("CreateArchivedBobWf", func() {
-		bobWf = (s.e().POST("/api/v1/workflows/argo").
+		bobWf := s.e().POST("/api/v1/workflows/argo").
 			WithBytes([]byte(`{
 				  "workflow": {
 					"metadata": {
@@ -1054,16 +1054,14 @@ func (s *ArgoServerSuite) TestWorkflowServiceListArchived() {
 					}
 				  }
 				}`)).
-			Expect().Status(200).JSON())
+			Expect().Status(200).JSON()
+		uidBobWf = bobWf.Path("$.metadata.uid").NotNull().String().Raw()
+		nameBobWf = bobWf.Path("$.metadata.name").NotNull().String().Raw()
 	})
-	var uidBobWf = bobWf.Path("$.metadata.uid").
-		NotNull().String().Raw()
-	var nameBobWf = bobWf.Path("$.metadata.name").
-		NotNull().String().Raw()
 
-	var aliceWf *httpexpect.Value
+	var uidAliceWf, nameAliceWf string
 	s.Run("CreateAlice", func() {
-		aliceWf = (s.e().POST("/api/v1/workflows/argo").
+		aliceWf := s.e().POST("/api/v1/workflows/argo").
 			WithBytes([]byte(`{
 				  "workflow": {
 					"metadata": {
@@ -1086,12 +1084,10 @@ func (s *ArgoServerSuite) TestWorkflowServiceListArchived() {
 					}
 				  }
 				}`)).
-			Expect().Status(200).JSON())
+			Expect().Status(200).JSON()
+		uidAliceWf = aliceWf.Path("$.metadata.uid").NotNull().String().Raw()
+		nameAliceWf = aliceWf.Path("$.metadata.name").NotNull().String().Raw()
 	})
-	var uidAliceWf = aliceWf.Path("$.metadata.uid").
-		NotNull().String().Raw()
-	var nameAliceWf = aliceWf.Path("$.metadata.name").
-		NotNull().String().Raw()
 
 	s.Given().When().
 		WaitForWorkflow(fixtures.ToBeArchived, metav1.ListOptions{FieldSelector: "metadata.name=" + nameBobWf}).
@@ -1215,9 +1211,9 @@ func (s *ArgoServerSuite) TestWorkflowServiceListArchived() {
 }
 
 func (s *ArgoServerSuite) TestWorkflowArchiveServiceList() {
-	var bobWf *httpexpect.Value
+	var uidBobWf, nameBobWf string
 	s.Run("CreateArchivedBobWf", func() {
-		bobWf = (s.e().POST("/api/v1/workflows/argo").
+		bobWf := s.e().POST("/api/v1/workflows/argo").
 			WithBytes([]byte(`{
 				  "workflow": {
 					"metadata": {
@@ -1240,16 +1236,14 @@ func (s *ArgoServerSuite) TestWorkflowArchiveServiceList() {
 					}
 				  }
 				}`)).
-			Expect().Status(200).JSON())
+			Expect().Status(200).JSON()
+		uidBobWf = bobWf.Path("$.metadata.uid").NotNull().String().Raw()
+		nameBobWf = bobWf.Path("$.metadata.name").NotNull().String().Raw()
 	})
-	var uidBobWf = bobWf.Path("$.metadata.uid").
-		NotNull().String().Raw()
-	var nameBobWf = bobWf.Path("$.metadata.name").
-		NotNull().String().Raw()
 
-	var aliceWf *httpexpect.Value
+	var uidAliceWf, nameAliceWf string
 	s.Run("CreateAlice", func() {
-		aliceWf = (s.e().POST("/api/v1/workflows/argo").
+		aliceWf := s.e().POST("/api/v1/workflows/argo").
 			WithBytes([]byte(`{
 				  "workflow": {
 					"metadata": {
@@ -1272,12 +1266,10 @@ func (s *ArgoServerSuite) TestWorkflowArchiveServiceList() {
 					}
 				  }
 				}`)).
-			Expect().Status(200).JSON())
+			Expect().Status(200).JSON()
+		uidAliceWf = aliceWf.Path("$.metadata.uid").NotNull().String().Raw()
+		nameAliceWf = aliceWf.Path("$.metadata.name").NotNull().String().Raw()
 	})
-	var uidAliceWf = aliceWf.Path("$.metadata.uid").
-		NotNull().String().Raw()
-	var nameAliceWf = aliceWf.Path("$.metadata.name").
-		NotNull().String().Raw()
 
 	s.Given().When().
 		WaitForWorkflow(fixtures.ToBeArchived, metav1.ListOptions{FieldSelector: "metadata.name=" + nameBobWf}).

